scripts/tests/twister_blackbox/conftest.py

```diff
@@ -43,3 +45,17 @@ def clear_log():
         handlers = getattr(logger, 'handlers', [])
         for handler in handlers:
             logger.removeHandler(handler)
+
+@pytest.fixture(autouse=True)
+def clear_out():
```

Comment:
I am not sure if this is a good approach. It interferes with the host environment quite some. It will remove all directories with `twister-out` and a user might be surprised with it (I know it is meant for the CI, however, user tend to run such tests locally, e.g. to reproduce some failures)
I think the better approach would be to modify the overall workflow. A fixture for it could create a "temp" directory as a "Before" section and yield it, twister could be called pointing directly to this dir as a location for outputs, and the whole dir could be removed in the "After" section. With such approach the tests won't interfere with user setup. 
BTW With the current approach the tests can be also influenced by an environment. twister creates `twister-out` dirs up to 99. If a user have too many of those already, the black_box results will be corrupted

Reply:
I have switched to creating a temp folder and using it with the `--outdir` option in twister to separate the blackbox output from the user's environment.